### PR TITLE
Adding in the terms handler so that generated apps inculde this by default

### PR DESCRIPTION
**Pull request by cam156**
This PR is dependant on #1119 

Also adding in a few additional changes from Sufia that seemed reasonable

This change should fix the issue that some people have been seeing when running tests in other gems:

```
   Failure/Error:
       json = solr_connection.get 'terms', params: { 'terms.fl' => term,
```

The solr config that is currently generated by active fedora does not include the terms configuration.

### CODE REVIEW

File: lib/generators/active_fedora/config/solr/templates/solr/config/solrconfig.xml [hunk 1]
```diff
@@ -102,6 +102,7 @@
        -->
         <str name="qf">
           id
+          active_fedora_model_ssi
```
Comment (jcoyne) [inline lib/generators/active_fedora/config/solr/templates/solr/config/solrconfig.xml]: what does this have to do with the terms handler?

**Comment by jcoyne** [comment]
This should be rebased now.

File: lib/generators/active_fedora/config/solr/templates/solr/config/solrconfig.xml (hunk 1 above)
Comment (cam156) [inline lib/generators/active_fedora/config/solr/templates/solr/config/solrconfig.xml]: @jcoyne nothing as far as I could tell, but it seemed reasonable...

**Comment by cam156** [comment]
@jcoyne rebased.  I got the additional changes, like active_fedora_model_ssi, from the Sufia configuration.  I am willing to remove them if you would like.

**Comment by jcoyne** [comment]
@cam156 Yes, I think the PR should be as narrowly scoped as possible.

**Comment by jcoyne** [comment]
If people have a poorly configured solr aren't they still going to have trouble when they try to do full text extraction?  That's a solr config thing that lives only in CurationConcerns/Sufia, right?   I wonder if we shouldn't make it easer for the correct configs to be created from Sufia, rather than make changes here.

File: lib/generators/active_fedora/config/solr/templates/solr/config/solrconfig.xml (hunk 1 above)
Comment (mjgiarlo) [inline lib/generators/active_fedora/config/solr/templates/solr/config/solrconfig.xml]: I wonder if this is cruft. `active_fedora_model_ssi` does not seem to be widely used; I could only find two references to that field in Sufia's dependencies:

 * https://github.com/projecthydra/curation_concerns/blob/master/spec/views/curation_concerns/base/_member.html.erb_spec.rb#L6
 * https://github.com/projecthydra/hydra-head/blob/master/hydra-core/spec/helpers/blacklight_helper_spec.rb#L23-L25

**Comment by cam156** [comment]
@jcoyne I thought about changing the configs just in Sufia.  It seemed like this issue was giving multiple people at multiple layers heartburn, I put in a root fix.
@jcoyne I'll remove the extras...

**Comment by mjgiarlo** [comment]
@cam156 :heart: :fire: --

**Comment by cam156** [comment]
@jcoyne or others... The extras have been removed.  Merge or Move?

**Comment by awead** [comment]
I vote merge.

**Comment by jcoyne** [comment]
What about the extraction handler?

**Comment by awead** [comment]
Meh. I can go either way, but the ticket is for terms handler.

**Comment by jcoyne** [comment]
Yeah, but we should either have a configuration that has all the needs of Sufia, or don't bother, right? Why have a configuration that only has some of the needs met?

**Comment by awead** [comment]
OK, that makes sense. The extraction is already a part of Solr, it's just a matter of turning it on, right? I just want to make sure that we're not requiring additional things to be downloaded/installed that ActiveFedora isn't using directly.

**Comment by jcoyne** [comment]
AFAIK, everything is already downloaded, we just need the correct config. Up until now I believe that's lived in CC/Sufia.

**Comment by mjgiarlo** [comment]
I support this PR, and wonder if it should also include the same change to `solr/config/solrconfig.xml` (or wipe out `solr/config/` entirely -- see #1125).
